torch/fx/graph.py

```diff
@@ -753,6 +753,39 @@ def generate_output(self, output_args):
         else:
             return super().generate_output(output_args)
 
+class _GraphSideTable:
+    """
+    Side table for the graph for the purpose of doing fast queries
+    """
+    def __init__(self):
+        self.table: Dict[Tuple[str, Optional[Target]], Dict[Node, None]] = defaultdict(dict)
+
+    def _key(self, node) -> Tuple[str, Optional[Target]]:
+        return (node.op, node.target if node.op == "call_function" else None)
+
+    def __contains__(self, node) -> bool:
+        return node in self.table[self._key(node)]
+
+    def insert(self, node: Node) -> None:
+        self.table[self._key(node)][node] = None
+
+    def remove(self, node: Node) -> None:
+        self.table[self._key(node)].pop(node)
+
+    def find_nodes(self, *, op: str, target: Optional['Target'] = None):
+        if op == "call_function":
+            assert target is not None
+            return sorted(dict(self.table[(op, target)]).keys())
+
+        if op in ("placeholder", "output"):
+            assert target is None
+
```

Comment:
I think it is fine to remove this assert, since the code below will handle it just fine.